tests/test_trajectory.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+
+import unittest
+from ddt import ddt, data, unpack
+
+import numpy as np
+
+import broni
+from broni.shapes.primitives import Cuboid, Sphere
+
+
+@ddt
+class TestTrajectory(unittest.TestCase):
+    def test_invalid_ctor_args_raise_exceptions(self):
+        # 3 columns are needed
+        with self.assertRaises(ValueError):
+            assert broni.Trajectory(
+                np.array([[0, 0], [0, 0]], dtype=float),
+                [2, 2],
+                coordinate_system="gse"
+            )
+
+        # 2 dim array is required
+        with self.assertRaises(ValueError):
+            assert broni.Trajectory(
+                np.array([0, 0, 0], dtype=float),
+                [2, 2, 2],
+                coordinate_system="gse"
+            )
+
+        # empty trajectory not supported
+        with self.assertRaises(ValueError):
+            assert broni.Trajectory(
+                np.array([[]], dtype=float),
+                [],
+                coordinate_system="gse"
+            )
+
+        # time-series too long
+        with self.assertRaises(ValueError):
+            assert broni.Trajectory(
+                np.array([[0, 0, 0], [0, 0, 0]], dtype=float),
+                [1, 2, 3],
+                coordinate_system="gse"
+            )
```

Comment:
I would write a function for each case so the name could describe the test and it would also ensure that all tests are run even if first case fails. 
I'm not 100% sure, but I think that here, if first with statement fails then it leaves the function.

Reply:
Agreed. I'm changing it.